patterns/creational/prototype.py

```diff
@@ -21,15 +21,19 @@
 Creates new object instances by cloning prototype.
 """
 
+from typing import Any, Dict
 
-class Prototype:
 
-    value = "default"
+class Prototype:
+    def __init__(self, value: str = "default", **attrs: Any) -> None:
+        self.value = value
+        self.__dict__.update(attrs)
 
-    def clone(self, **attrs):
+    def clone(self, **attrs: Any) -> None:
         """Clone a prototype and update inner attributes dictionary"""
         # Python in Practice, Mark Summerfield
-        obj = self.__class__()
+        # copy.deepcopy can be used instead of next line.
+        obj = self.__class__(**self.__dict__)
```

Comment:
It is possible like below if this prototype is just copy the `value` into clones.
```
obj = self.__class__(value=self.value)
```